packages/ai-jsx/src/batteries/use-tools.tsx

```diff
@@ -43,7 +45,7 @@ function ChooseTools(props: Pick<UseToolsProps, 'tools' | 'userData' | 'query'>)
         If none of the tools seem appropriate, or the user data doesn't have the necessary context to use the tool the
         user needs, respond with `null`.
       </SystemMessage>
-      <UserMessage>Generate a JSON response for this query: {props.query}</UserMessage>
+      <UserMessage>Generate a JSON response for this query: {props.children}</UserMessage>
```

Comment:
I feel like the interface for the prompt-engineered version and the function-call version have diverged. For example, if you give it a `SystemMessage`, it will be ignored. I haven't used the following, but maybe something like this would work:
```suggestion
      <UserMessage>Generate a JSON response for the query below</UserMessage>
      {props.children}
```

Reply:
You might need to memoize here too?